docs: add tip on Kubernetes resources (#22977)
gravitational/teleport: docs/pages/kubernetes-access/controls.mdx

regular expression using Go's `re2` syntax (see the `re2`
[README](https://github.com/google/re2/wiki/Syntax)).

<Notice type="tip" >
  For a user to have Kubernetes access `kubernetes_groups` or `kubernetes_users`

>   must also be assigned within a role to a user. `kubernetes_resources` configuration

Reviewer: I read this a few times and the wording feels off. Maybe something like this instead?

> For a user to access Kubernetes clusters, they must be assigned a role
> that contains `kubernetes_groups` or `kubernetes_users`.

Author: please see update

>   must also be assigned within a role to a user.  `kubernetes_resources` configuration

Reviewer 2: Old habits die hard.
```suggestion
  must also be assigned within a role to a user. `kubernetes_resources` configuration
```

  requires the users or groups within the Kubernetes clusters have access to the pods

>   and does not grant additional access if the Kubernetes role does not.

Reviewer: I don't understand what this last part of the sentence is trying to say. Is there a simpler way to phrase this?

regular expression using Go's `re2` syntax (see the `re2`
[README](https://github.com/google/re2/wiki/Syntax)).

<Notice type="tip" >
  For a user to have Kubernetes access they must be asssigned a Teleport role that
  contains `kubernetes_groups` or `kubernetes_users`. The groups or user assigned
  must be associated to roles with access to the pods within that Kubernetes
  cluster. Teleport does not alter those roles to allow or deny access.
  Teleport confirms access based on the Teleport `pod` role settings at 

>   the Teleport Kubernetes service and then by the Kubernetes cluster RBAC.

Reviewer 2: ```suggestion
  the Teleport Kubernetes service, and then by the Kubernetes cluster RBAC.
```

>   For a user to have Kubernetes access they must be asssigned a Teleport role that
>   contains `kubernetes_groups` or `kubernetes_users`. The groups or user assigned
>   must be associated to roles with access to the pods within that Kubernetes
>   cluster. Teleport does not alter those roles to allow or deny access.
>   Teleport confirms access based on the Teleport `pod` role settings at 
>   the Teleport Kubernetes service, and then by the Kubernetes cluster RBAC.

Reviewer 3: ```suggestion
For a user to access a pod named in a role's `kubernetes_resources` field, the user
must be assigned a Teleport role that contains at least one value within 
`kubernetes_groups` or `kubernetes_users`. Teleport does not alter Kubernetes 
roles to allow or deny access. Read the next section for an explanation of how the 
Kubernetes Service evaluates Teleport roles in order to allow or deny access to 
pods in a cluster.
```

I think this text is trying to say too much in a compressed space. Since there is a more detailed explanation of how the Teleport Kubernetes Service evaluates roles in the next section, I've suggested putting the key points here and recommending that the reader continue to the next section.

I wasn't sure what "Teleport does not alter these roles" meant in the original text, so feel free to correct the relevant sentence.
